go-app-chw.js

```
@@ -921,7 +968,23 @@ go.app = function() {
                         'woman will now receive messages to encourage her ' +
                         'to register at her nearest clinic.'),
 
-                next: 'states:start'
+                next: 'states:start',
+                events: {
+                    'state:enter': function() {
+                        var built_json = go.utils.build_json_doc(self.contact, self.user, "pre-registration");
+                        return go.utils.jembi_json_api_call(built_json, self.im)
+                            .then(function(result) {
+                                if (result.code >= 200 && result.code < 300){
```

I'd suggest you fail fast here and not do the logging or always `return true`? Your fixture will let you test whatever you want.